### PR TITLE
[Build] Asan wheel test

### PR DESCRIPTION
**Pull request by rkooo567**
## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by simon-mo** [comment]
lmk if you need a bigger box for these, you can pick any aws instance type

**Comment by rkooo567** [comment]
@simon-mo sounds good! I think for the initial iteration, we don't need that, but we will eventually need bigger machines to run asan tests.

**Review by simon-mo — APPROVED** [review]

File: ci/travis/ci.sh [hunk 1]
```diff
@@ -297,10 +297,16 @@ _bazel_build_before_install() {
   # NOTE: Do not add build flags here. Use .bazelrc and --config instead.
 
   # Build in debug mode if RAY_DEBUG_BUILD=1
-  if [ -z "${RAY_DEBUG_BUILD-}" ] || [ "${RAY_DEBUG_BUILD}" -ne "1" ]; then
+  if [ -z "${RAY_DEBUG_BUILD-}" ]; then
```
Comment (scv119) [inline ci/travis/ci.sh:300]: image

**Review by scv119 — APPROVED** [review]

**Comment by rkooo567** [comment]
mac failure seems unrelated

**Pull request merged**